app/Http/Controllers/QuizController.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Http\Requests\StoreQuizResultRequest;
+use App\Models\Quiz;
+use Illuminate\Contracts\View\View;
+use Illuminate\Support\Facades\Auth;
+
+class QuizController extends Controller
+{
+    public function show(Quiz $quiz): View
+    {
+        $questions = $quiz->questions;
+
+        return view('quizzes.show', [
+            'quiz' => $quiz,
+            'title' => $quiz->title,
+            'questions' => $questions,
+        ]);
+    }
+
+    public function storeQuizResult(StoreQuizResultRequest $request, Quiz $quiz): View
+    {
+        $data = $request->validated();
+        $user = Auth::user();
+
+        $numberOfQuestions = 0;
+        $quizScore = 0;
+
+        foreach ($quiz->questions as $question) {
+            $numberOfQuestions++;
+            if (count(array_intersect($question->getCorrectChoices(), $data['question_'.$question->id])) >= $question->minimum_choices) {
+                $quizScore++;
+            }
+        }
+        $quizResults = $quizScore / $numberOfQuestions >= $quiz->minimum_score;
+        $quizUser = $user->quizzes->where('id', $quiz->id)->first()->pivot ?? null;
```

Comment:
The null coalescing is not needed here. The `??` means that if the left hand side is null return what is on the right hand side. What you are doing here in effect is checking if the `$user->quizzes->where('id', $quiz->id)->first()->pivot` is null then return null. So you could just remove it and you'll get the same value back.

```suggestion
        $quizUser = $user->quizzes->where('id', $quiz->id)->first()->pivot;
```

I'm not sure if I'm reading the logic here wrong or not, but it would also seem that you could simplify this because you are finding a specific quiz by id, you could use `find` and shouldn't need to call `first` as you are only getting one.

```suggestion
        $quizUser = $user->quizzes->find($quiz)->pivot;
```

It probably doesn't matter, but you might be able to go the opposite way. I wonder if this reads better because we are in the Quiz controller? Let me know what you think.

```suggestion
        $quizUser = $quiz->users()->find($user)->pivot;
```